### ISSUE TITLE
running demo on CUTE80 with SATRN

### ISSUE THREAD

**Issue by gussmith** [Media-Smart/vedastr#15]
I am running the demo on the Curved Text of CUTE80: 
`python tools/demo.py configs/small_satrn.py pretrained/small-SATRN/Small-SATRN.pth ../../data/CUTE80-Curved-Text/CUTE80`

Do the images need to be pre-processed in some ways before being put into the directory? Like resized? 
What are the constraints? 

Currently, I using the raw images of CUTE80 like image001.jpg "Ronaldo 7" 650x440 pixels
and so far the decoding going over a few of the images is not close. 
Any suggestions? Thanks.

**Comment by ChaseMonsterAway** [comment]
Transforms are done in ``demo.py``.  The demo will close until decoding all the images under the directory.

**Comment by gussmith** [comment]
I reread the paper, and saw that the authors state: `"Input images are resized to 32 x 100 both during training and testing following common practice." `
Should we assume then that the pre-trained model small-SATRN was trained the same way? 
I will try resizing the input images, unless that's already done in the demo code?

**Comment by gussmith** [comment]
I stepped through the code and yes, the original images in CUTE80 (some like 3600 x 2800) get turned to gray, resized to 100 x 32 and normalized.

**Comment by ChaseMonsterAway** [comment]
Yes, transforms are done in ``demo.py``

**Comment by gussmith** [comment]
Yes, the issue is that when I use my own separate set of images from CUTE80, the performance is not good. But when I ran with the code of yesterday, I would get acc ~ 0.85 using test.py. 

Note that now, with the updated code, the results are not  good when tested on CUTE80, it looks like the last commits reduced the accuracy: 
```
vedastr$ python tools/test.py configs/small_satrn.py pretrained/small-SATRN/Small-SATRN.pth 
2020-07-29 12:16:08,842 - INFO - Use GPU 0
2020-07-29 12:16:08,843 - INFO - Set cudnn deterministic False
2020-07-29 12:16:08,843 - INFO - Set cudnn benchmark True
2020-07-29 12:16:08,843 - INFO - Set seed 1111
2020-07-29 12:16:08,843 - INFO - Build model
2020-07-29 12:16:08,846 - INFO - GResNet init weights
2020-07-29 12:16:09,142 - INFO - TransformerEncoder init weights
2020-07-29 12:16:09,596 - INFO - TransformerDecoder init weights
2020-07-29 12:16:09,686 - INFO - TransformerHead init weights
2020-07-29 12:16:11,304 - INFO - current dataset length is 288 in ./data/data_lmdb_release/evaluation/CUTE80
2020-07-29 12:16:11,304 - INFO - Load checkpoint from pretrained/small-SATRN/Small-SATRN.pth
2020-07-29 12:16:11,420 - INFO - Start testing
2020-07-29 12:16:11,420 - INFO - test info: {'sensitive': False, 'character': '0123456789abcdefghijklmnopqrstuvwxyz'}
2020-07-29 12:16:18,444 - INFO - Test, acc 0.1424, edit 0.23049960221608767
```

"acc" is for "accuracy"
but what is "edit" ?

**Comment by gussmith** [comment]
reverting to a previous commit to get it to run: 
`fix:converter`
`git checkout f4a30353132c018c21eec0f4c9a4e084254679c1
`
then the accuracy is back to what is was on CUTE80: 

`2020-07-29 09:03:01,760 - INFO - Test, acc 0.8467, edit 0.9275427916694391`

**Comment by ChaseMonsterAway** [comment]
> Yes, the issue is that when I use my own separate set of images from CUTE80, the performance is not good. But when I ran with the code of yesterday, I would get acc ~ 0.85 using test.py.
> 
> Note that now, with the updated code, the results are not good when tested on CUTE80, it looks like the last commits reduced the accuracy:
> 
> ```
> vedastr$ python tools/test.py configs/small_satrn.py pretrained/small-SATRN/Small-SATRN.pth 
> 2020-07-29 12:16:08,842 - INFO - Use GPU 0
> 2020-07-29 12:16:08,843 - INFO - Set cudnn deterministic False
> 2020-07-29 12:16:08,843 - INFO - Set cudnn benchmark True
> 2020-07-29 12:16:08,843 - INFO - Set seed 1111
> 2020-07-29 12:16:08,843 - INFO - Build model
> 2020-07-29 12:16:08,846 - INFO - GResNet init weights
> 2020-07-29 12:16:09,142 - INFO - TransformerEncoder init weights
> 2020-07-29 12:16:09,596 - INFO - TransformerDecoder init weights
> 2020-07-29 12:16:09,686 - INFO - TransformerHead init weights
> 2020-07-29 12:16:11,304 - INFO - current dataset length is 288 in ./data/data_lmdb_release/evaluation/CUTE80
> 2020-07-29 12:16:11,304 - INFO - Load checkpoint from pretrained/small-SATRN/Small-SATRN.pth
> 2020-07-29 12:16:11,420 - INFO - Start testing
> 2020-07-29 12:16:11,420 - INFO - test info: {'sensitive': False, 'character': '0123456789abcdefghijklmnopqrstuvwxyz'}
> 2020-07-29 12:16:18,444 - INFO - Test, acc 0.1424, edit 0.23049960221608767
> ```
> 
> "acc" is for "accuracy"
> but what is "edit" ?

Edit is edit distance. You can reference [here](https://en.wikipedia.org/wiki/Edit_distance) for edit distance.
Could you provide the commit id ?

**Comment by ChaseMonsterAway** [comment]
This is caused by the test: transforms (label is not lower) in config file. I have fixed it.
